src/alchemlyb/parsing/namd.py

```diff
@@ -9,14 +9,71 @@
 
 k_b = R_kJmol * kJ2kcal
 
+def get_lambdas(fep_files):
+    """Retrieves all lambda values included in the FEP files provided.
+    
+    We have to do this in order to tolerate truncated and restarted fepout files.
+    The IDWS lambda is not present at the termination of the window, presumably
+    for backwards compatibility with ParseFEP and probably other things.
+
+    For a given lambda1, there can be only one lambda_idws.
+
+    Parameters
+    ----------
+    fep_files: str or list of str
+        Path(s) to fepout files to extract data from.
+
+    Returns
+    -------
+    List of floats, or None if there is more than one lambda_idws for each lambda1.
+    """
+
+    lambda_fwd_map, lambda_bwd_map = {}, {}
+
+    for fep_file in sorted(fep_files):
+        with anyopen(fep_file, 'r') as f:
+            for line in f:
+                l = line.strip().split()
+                if l[0] not in ['#NEW', '#Free']:
+                    continue
+
+                # We might not have a #NEW line so make the best guess
+                if l[0] == '#NEW':
+                    lambda1, lambda2 = float(l[6]), float(l[8])
+                    lambda_idws = float(l[10]) if 'LAMBDA_IDWS' in l else None
+                elif l[0] == '#Free':
+                    lambda1, lambda2, lambda_idws = float(l[7]), float(l[8]), None
+
+                # Make sure the lambda2 values are consistent
+                if lambda1 in lambda_fwd_map and lambda_fwd_map[lambda1] != lambda2:
+                    print(f'namd.py: get_lambdas: Error: fwd: lambda1 {lambda1} has lambda2 {lambda_fwd_map[lambda1]} but it should be {lambda2}')
+                    return None
+
+                lambda_fwd_map[lambda1] = lambda2
+
+                # Make sure the lambda_idws values are consistent
+                if lambda_idws is not None:
+                    if lambda1 in lambda_bwd_map and lambda_bwd_map[lambda1] != lambda_idws:
+                        print(f'namd.py: get_lambdas: Error: bwd: lambda1 {lambda1} has lambda_idws {lambda_bwd_map[lambda1]} but it should be {lambda_idws}')
+                        return None
```

Comment:
My concern is that returning `None` will silently discard data. It's cleaner to fail and have the user fix things.